### PR TITLE
feat: implement List#add

### PR DESCRIPTION
**Pull request by lilsweetcaligula**
Issue ref. [LFRB9IeI/387-api-list-feature](https://trello.com/c/LFRB9IeI/387-api-list-feature)

Depends on PRs #72 #73

### CODE REVIEW

File: functions/models/list/add.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import admin from 'firebase-admin';
+
+import { IList } from '../../schemas/list';
+
+async function add(
+  record: IList,
+  dbOverride?: admin.firestore.Firestore
+): Promise<{ id: string }> {
+  const db = dbOverride || admin.firestore();
```
Comment (lilsweetcaligula) [inline functions/models/list/add.ts]: I am having second thoughts if calling `admin.firestore()` each time like this is okay.

Documentation suggests that it simply "gets" the client, but I fear "get" can mean many things (including making extra connections which come at a monetary cost):
https://firebase.google.com/docs/reference/admin/node/admin.firestore#firestore

If this is a problem, we can call `admin.firestore()` only once in `/models/list.ts` (to be submitted), and simply pass it to all modules like `models/list/add.ts`. We can then make the `dbOverride` parameter required.

File: functions/models/list/add.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import admin from 'firebase-admin';
+
+import { IList } from '../../schemas/list';
+
+async function add(
+  record: IList,
+  dbOverride?: admin.firestore.Firestore
```
Comment (lilsweetcaligula) [inline functions/models/list/add.ts]: The `dbOverride` param is used in tests, where a Firestore mock is passed and the logic of the model method is tested.

File: functions/models/list/add.ts (hunk 1 above)
Comment (nazreen) [reply functions/models/list/add.ts]: by 'each time' do you mean in each of this method folders? i.e. in add, get, list

or in each entity's model? i.e. promise, list, politician

File: functions/models/list/add.ts (hunk 2 above)
Comment (nazreen) [reply functions/models/list/add.ts]: noted

**Pull request merged**